modules/rostests/apitests/shell32/LocaleTests.cpp

```diff
@@ -0,0 +1,290 @@
+}
+
+static BOOL CALLBACK find_locale_id_callback(
+    _In_ HMODULE hModule, _In_ LPCWSTR type, _In_ LPCWSTR name, _In_ LANGID lang, _In_ LPARAM lParam)
+{
+    langs.insert(lang);
+    return TRUE;
+}
+
+static void SetLocale(_In_ LCID lcid)
+{
+    SetThreadLocale(lcid);
+    SetThreadUILanguage(lcid);
+    curLcid = lcid;
+}
+
+static void TEST_NumParts(void)
+{
+    for (auto i = parts.begin(); i != parts.end(); i++)
+    {
+        E_MODULE m = i->second.eModule;
+
+        if (!mod[m])
+        {
+            skip("No module for test #%d\n", i->first);
+            continue;
+        }
+
+        WCHAR szBuffer[MAX_PATH];
+
+        LoadStringW(mod[m], i->second.id, szBuffer, _countof(szBuffer));
+        ok(i->second.nParts == CountParts(szBuffer), "Locale 0x%lX: Num parts mismatch #%d - expected %lu, got %lu\n",
+           curLcid, i->first, i->second.nParts, CountParts(szBuffer));
+    }
+}
+
+static BOOL LoadPart(_In_ PART* p, _Out_ LPWSTR str, _In_ SIZE_T size)
+{
+    auto s = parts[p->Num];
+    E_MODULE m = s.eModule;
+
+    if (!mod[m])
+        return FALSE;
+
+    WCHAR szBuffer[MAX_PATH];
+    LPWSTR szPart;
+    SIZE_T len;
+
+    LoadStringW(mod[m], s.id, szBuffer, _countof(szBuffer));
+    szPart = GetPart(szBuffer, p->Idx, &len);
+    StringCchCopyNW(str, size, szPart, len);
+
+    return TRUE;
+}
+
+static void TEST_PartMatches(void)
+{
+    for (SIZE_T i = 0; i < _countof(PartMatches); i++)
+    {
+        WCHAR szP1[MAX_PATH], szP2[MAX_PATH];
+
+        if (!LoadPart(&PartMatches[i].p1, szP1, _countof(szP1)))
+        {
+            skip("No module for match test #%lu pair 1\n", i);
+            continue;
+        }
+
+        if (!LoadPart(&PartMatches[i].p2, szP2, _countof(szP2)))
+        {
+            skip("No module for match test #%lu pair 2\n", i);
+            continue;
+        }
+
+        ok(wcscmp(szP1, szP2) == 0, "Locale 0x%lX: Mismatching pairs #%u:%u / #%u:%u '%S' vs. '%S'\n",
+           curLcid, PartMatches[i].p1.Num, PartMatches[i].p1.Idx, PartMatches[i].p2.Num, PartMatches[i].p2.Idx, szP1, szP2);
+    }
+}
+
+static void TEST_LocaleTests(void)
+{
+    // Initialization
+    WCHAR szOldDir[MAX_PATH], szBuffer[MAX_PATH];
+    GetCurrentDirectoryW(_countof(szOldDir), szOldDir);
+
+    std::map<E_MODULE, LPCWSTR> lib;
+
+    GetModuleFileNameW(NULL, szBuffer, _countof(szBuffer));
+    LPWSTR pszFind = StrStrW(szBuffer, L"modules\\rostests\\apitests");
+    if (pszFind)
+    {
+        WCHAR szNewDir[MAX_PATH];
+
+        StringCchCopyNW(szNewDir, _countof(szNewDir), szBuffer, pszFind - szBuffer);
+        SetCurrentDirectoryW(szNewDir);
+
+        lib = {
+            {shell32, L"dll\\win32\\shell32\\shell32.dll"},
+            {userenv, L"dll\\win32\\userenv\\userenv.dll"},
+            {syssetup, L"dll\\win32\\syssetup\\syssetup.dll"},
+            {mmsys, L"dll\\cpl\\mmsys\\mmsys.cpl"},
+            {explorer_old, L"modules\\rosapps\\applications\\explorer-old\\explorer_old.exe"},
+        };
```

Comment:
error C2679: binary '=': no operator found which takes a right-hand operand of type 'initializer list' (or there is no acceptable conversion)

(and ditto on line 241)

Reply:
[modules.h.zip](https://github.com/user-attachments/files/16173892/modules.h.zip)

Use this. You can manually add them.